service-fabric-secure-cluster-5-node-1-nodetype-wad/azuredeploy.json

```diff
@@ -0,0 +1,761 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "Location of the Cluster"
+      }
+    },
+    "clusterName": {
```

Comment:
Sure we can merge today but I think it's important to take the time to recognize the potential point of failures and address them since the friction and cost of failing and having to enter parameters and deploy again is pretty high. I see lines 134, 135 as two potential points of failure since this parameter is being used as is for things that need to be globally unique. Also storage account has another limitation that it needs to have all lower case. So any upper case letters in cluster name will cause the deployment to fail.

    "dnsName": "[parameters('clusterName')]", 
    "vmStorageAccountName": "[concat(parameters('clusterName'),'1')]", 
 
The simplest solution is to give the user an indication to enter a unique value for this parameter. Doing that by renaming this parameter to "uniqueClusterName" should be good. Then you could do

    "storageAccountName" = "[toLower(parameters('uniqueClusterName'))]"

in variables section to prevent failures due to upper case.

But the truly ideal solution which requires more work would be to concat this parameter value with the 13 character output of the uniqueString() function but then you need to limit the input for clusterName to 11 characters since storage accounts can only be a max of 24 characters. This template is a good example of how to that https://github.com/Azure/azure-quickstart-templates/blob/bfb0c40965969a4884ccee65ff75c7ee13e3195b/301-2fe-linux-lb80-ssh-1be-win-nsg-rdp-datadisk-ssd/azuredeploy.json#L90

    "newStorageAccountName": "[concat('pst', uniqueString(resourceGroup().id))]",